src/typed-method-types/workflows/triggers/scheduled.ts

```diff
@@ -0,0 +1,94 @@
+import { BaseTrigger, RequiredInputs, TriggerTypes } from "./mod.ts";
+
+const SCHEDULE_FREQUENCY = {
+  Daily: "daily",
+  Weekly: "weekly",
+  Monthly: "monthly",
+  Yearly: "yearly",
+} as const;
+
+const WEEKDAYS = {
+  Monday: "Monday",
+  Tuesday: "Tuesday",
+  Wednesday: "Wednesday",
+  Thursday: "Thursday",
+  Friday: "Friday",
+  Saturday: "Saturday",
+  Sunday: "Sunday",
+} as const;
+
+type Frequency = typeof SCHEDULE_FREQUENCY[keyof typeof SCHEDULE_FREQUENCY];
+
+type BaseFrequencyType = {
+  /** @description How often the trigger will activate */
+  type: Frequency;
+  /** @description The days of the week the trigger should activate on (not available for daily triggers) */
+  on_days?: (typeof WEEKDAYS[keyof typeof WEEKDAYS])[];
+  /** @description How often the trigger will repeat, respective to the frequency type */
+  repeats_every?: number;
+  /**
+   * @description The nth week of the chosen frequency type (not available for daily or weekly triggers)
+   * @example The 3rd week of the month */
+  on_week_num?: number;
+};
+
+type DailyFrequencyType =
+  & {
+    /** @description How often the trigger will activate */
+    type: typeof SCHEDULE_FREQUENCY.Daily;
+  }
+  & Pick<BaseFrequencyType, "repeats_every">;
+
+type WeeklyFrequencyType = {
+  /** @description How often the trigger will activate */
+  type: typeof SCHEDULE_FREQUENCY.Weekly;
+} & Pick<BaseFrequencyType, "on_days" | "repeats_every">;
+
+type MonthlyOrYearlyFrequencyType = {
```

Comment:
I think we can have separate type for monthly & yearly. This type looks good for monthly but for yearly we can optionally have `repeats_every` in frequency type (similar to daily frequency).